### PR TITLE
Migrated language settings screen from XML to Jetpack compose

### PR DESCRIPTION
**Pull request by Saifuddin53**
### Contributor checklist

-   [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
-   [] I have tested my code with the `./gradlew lintKotlin detekt test` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Android/blob/main/CONTRIBUTING.md#testing)

---

### Description

The pull request moves the language settings page to Jetpack Compose and has refactored the settings composable to use the reusable components that were created in this PR

### Screenshots
Settings: 

| Before | After |
|---|---|
| settings | settings_after

Language Settings:

| Before | After |
|---|---|
| lansettings | lansettings_after

-   #ISSUE_NUMBER
#253

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks for the quick work here, @Saifuddin53! I'll get to my review this weekend 😊 Great to have you on the project :)

**Comment by andrewtavis** [comment]
Had a moment to check the UI and it's all looking great 😊 Thanks @Saifuddin53 and @angrezichatterbox for working towards such a professional and technologically sound application :) @angrezichatterbox, do you want to do a quick check of the code and then we can bring this in?

**Review by angrezichatterbox — COMMENTED** [review]
@Saifuddin53, your work on refactoring the compose codebase into reusable and maintainable components is truly impressive. The page you’ve created is also amazing. I hope we’ll see you around solving more issues in the future.

There are a few linting issues that could be resolved by running `./gradlew formatKotlin`.

File: app/src/main/java/be/scri/ui/models/ScribeItem.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    data class CustomItem(
+        override val title: String,
+        override val desc: String,
+        val customAction: (Any?) -> Unit
+    ) : ScribeItem(title, desc)
+}
+
+
+enum class ItemType {
```
Comment (angrezichatterbox) [inline app/src/main/java/be/scri/ui/models/ScribeItem.kt:36]: This is currently not being used right and could be removed or the codebase could be modified to use them.

Comment (Saifuddin53) [reply app/src/main/java/be/scri/ui/models/ScribeItem.kt]: The enum class is no more of use and the other two data classes can be used in the future when we migrate the About page to compose

**Review by angrezichatterbox — APPROVED** [review]
Everything looks Great now :)

**Review by andrewtavis — APPROVED** [review]
Thank you both for the amazing collaboration here for a great result! 😊 Looking forward to the next issue, @Saifuddin53 🚀

**Pull request merged**